R/drive_search.R

```diff
@@ -47,6 +52,13 @@ drive_search <- function(pattern = NULL, ..., verbose = TRUE) {
     params$fields <- paste0("files/", .drive$default_fields, collapse = ",")
   }
 
+  if (!is.null(type)) {
+    mime_type <- drive_mime_type(type, verbose = verbose)
+    if (!is.null(mime_type)) {
+    params$q <- glue::collapse(c(params$q, paste0("mimeType = '", mime_type,"'")),
+                               sep = " and ")
+    }
+  }
```

Comment:
By picking the right functions, you can probably make this unconditional on whether `mime_type = NULL`. I.e., I think you could use `glue::glue()` instead of the `paste0()` and it will return `character()` if `mime_type = NULL`. Then probably `paste()` instead of `glue::collapse()` to combine `q` and formatted MIME type.